drivers/sensor/sensor_shell.c

```diff
@@ -470,6 +643,8 @@ SHELL_STATIC_SUBCMD_SET_CREATE(sub_sensor,
 			cmd_sensor_attr_get, 2, 255),
 	SHELL_COND_CMD(CONFIG_SENSOR_INFO, info, NULL, SENSOR_INFO_HELP,
 			cmd_get_sensor_info),
+	SHELL_CMD_ARG(trig, &dsub_device_name, SENSOR_TRIG_HELP, cmd_trig_sensor,
```

Comment:
Reusing dsub_device_name will get you the autocomplete for the `get` command, you'll need to create a new one with the same pattern so after you autocomplete the device name you'll autocomplete to `on/off` then to the attribute names.

Reply:
Okay, I thought this would just autosuggest device names but now wrote a full chain of helper to fill out the entire `sensor trig` command.